plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultQueryBuilder.java

```diff
@@ -317,9 +328,20 @@ else if (javaType == Slice.class) {
     public CallableStatement callProcedure(JdbcClient client, ConnectorSession session, Connection connection, ProcedureQuery procedureQuery)
             throws SQLException
     {
+        setClientInfo(connection, "ApplicationName", "Trino-" + session.getQueryId());
         return connection.prepareCall(procedureQuery.query());
     }
 
+    private void setClientInfo(Connection connection, String key, String value)
+    {
+        try {
+            connection.setClientInfo(key, value);
```

Comment:
Rather than replacing the current value, can you append to the current value? That way, if there is some value already present, we won't overwrite it.

Reply:
this is a generic method and generic method doesn't know whether appending makes sense.

for "application name" it probably doesn't